test/colab/colab-test.mk

```diff
@@ -0,0 +1,31 @@
+.PHONY: colab-test-internal
+colab-test-internal: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-pull
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-internal-local
+colab-test-internal-local: $(TEST_TOKEN_FILE) $(COLAB_TEST_HOME)/test_import_login_init.ipynb colab-test-build
+	$(DOCKER_RUN)
+
+.PHONY: colab-test-pull
+colab-test-pull:
+	@docker pull $(DOCKER_IMAGE_NAME)
+
+.PHONY: colab-test-build
+colab-test-build: $(COLAB_IMAGE_BUILD_STAMP)
+$(COLAB_IMAGE_BUILD_STAMP): $(COLAB_TEST_HOME)/requirements-fixed.txt test/colab/Dockerfile
+	@docker build -t $(DOCKER_IMAGE_NAME) -f test/colab/Dockerfile .
+	@touch $(COLAB_IMAGE_BUILD_STAMP)
+
+.DELETE_ON_ERROR: $(COLAB_TEST_HOME)/requirements-fixed.txt $(COLAB_TEST_HOME)/test_import_login_init.ipynb
+
+$(COLAB_TEST_HOME)/requirements-fixed.txt: test/colab/requirements.txt
+	@mkdir -p $(COLAB_TEST_HOME)
+# Fix requirements.txt to make them compatible with running on M1 mac/linux, outside of colab itself.
+	@./test/colab/fix-requirements.sh
+
+$(COLAB_TEST_HOME)/test_import_login_init.ipynb: test/colab/test_import_login_init.ipynb
+	@mkdir -p $(COLAB_TEST_HOME)
+# VSC saves notebooks with `null` in place of `0` for `execution_count`, e.g. `"execution_count": null,`.\
+`nbconvert` crashes on these, so we fix that here. 
+	@cp test/colab/test_import_login_init.ipynb $(COLAB_TEST_HOME)/test_import_login_init.ipynb
+	@sed -i'' -e "s/\"execution_count\": null,/\"execution_count\": 0,/g" $(COLAB_TEST_HOME)/test_import_login_init.ipynb
```

Comment:
I think we want that in `make format`, not when we run the tests as this notebook is committed in the code.

Reply:
No, we create a copy of the notebook under the `build` folder, and format that notebook, so this doesn't get committed.

Reply:
But when I checked out your branch and edited the notebook I got a diff with that exact same problem.

Reply:
Can no longer reproduce this.